airflow/providers/common/sql/hooks/sql.py

```diff
@@ -218,8 +217,9 @@ def get_pandas_df_by_chunks(self, sql, parameters=None, *, chunksize, **kwargs):
                 "'apache-airflow-providers-common-sql[pandas]'."
             )
 
-        with closing(self.get_conn()) as conn:
-            yield from psql.read_sql(sql, con=conn, params=parameters, chunksize=chunksize, **kwargs)
+        yield from psql.read_sql(
+            sql, con=self.get_sqlalchemy_engine(), params=parameters, chunksize=chunksize, **kwargs
+        )
```

Comment:
The same here: create connection -> close it (context manger) -> yielding chunks -> dispose engine